frontend/src/api/Socket.ts

```diff
@@ -3,14 +3,52 @@ import Stomp from 'stompjs';
 
 let stompClient:any = null;
 
-export const connect = (endpoint:string) => {
-  const socket:any = new SockJS(endpoint);
-  stompClient = Stomp.over(socket);
-  stompClient.connect();
+type User = {
+  nickname: string;
+}
+
+// Variable to hold the current connected state.
+let connected = false;
+
+// Create constants for the subscription and send message URLs.
+const SUBSCRIBE_URL:string = '/api/v1/socket/subscribe-user';
+const SEND_USER_URL:string = '/api/v1/socket/user';
+
+export const sendUser = (nickname:string) => {
+  if (connected) {
+    const nicknameInput:string = (nickname !== '') ? nickname : 'Anonymous';
```

Comment:
nit: Idk if this actually works or not but maybe try simplifying this to `nickname || 'Anonymous'`

Reply:
Ah, yeah that's smart -- it worked, thank you!